components/user_settings/display/index.ts

```diff
@@ -54,6 +54,8 @@ function mapStateToProps(state: GlobalState) {
         channelDisplayMode: get(state, Preferences.CATEGORY_DISPLAY_SETTINGS, Preferences.CHANNEL_DISPLAY_MODE, Preferences.CHANNEL_DISPLAY_MODE_DEFAULT),
         messageDisplay: get(state, Preferences.CATEGORY_DISPLAY_SETTINGS, Preferences.MESSAGE_DISPLAY, Preferences.MESSAGE_DISPLAY_DEFAULT),
         collapseDisplay: get(state, Preferences.CATEGORY_DISPLAY_SETTINGS, Preferences.COLLAPSE_DISPLAY, Preferences.COLLAPSE_DISPLAY_DEFAULT),
+        collapsedReplyThreadsAllowUserPreference: isCollapsedThreadsAllowed(state) && getConfig(state).CollapsedThreads as string !== 'always_on',
```

Comment:
nit: 'always_on' as a constant

Reply:
CollapsedThreads is also backed by explicit string literals, but we could possibly change it to an enum. I'll go ahead and do that, but maybe I'll queue up the topic for the next guild meeting? (Constants, enums, string literal unions, etc.)